src/components/people/PersonRoleList/RoleItem.tsx

```diff
@@ -0,0 +1,92 @@
+    PersonRole,
+    useCurrentUser,
+    useFusionContext,
+} from '@equinor/fusion';
+import { Switch } from '@equinor/fusion-components';
+import React, { useCallback, useState } from 'react';
+import * as styles from './styles.less';
+import { HttpClientRequestFailedError } from '@equinor/fusion/lib/http/HttpClient';
+import classNames from 'classnames';
+
+export type RoleSwitchProps = {
+    role: PersonRole;
+    showSwitch: boolean;
+};
+
+const expiresIn = (activeTo: string) => {
+    const activeToDate = new Date(activeTo).getTime();
+    const now = new Date().getTime();
+
+    if (now > activeToDate) {
+        return 'Expired';
+    }
+
+    return `Expires in ${Math.ceil(Math.abs(activeToDate - now) / 36e5)} hours`;
+};
+
+const RoleItem: React.FC<RoleSwitchProps> = ({ role, showSwitch }: RoleSwitchProps) => {
+    const currentUser = useCurrentUser();
+    const { peopleContainer } = useFusionContext();
+    const [isActive, setIsActive] = useState<boolean>(role.isActive);
+    const [errorMessage, setErrorMessage] = useState<string>('');
+
+    const toggleRoleStatus = useCallback(
+        async (clickedRole: PersonRole) => {
+            if (!currentUser) {
+                return;
+            }
+            setIsActive((previous) => !previous);
+            try {
+                await peopleContainer.setRoleStatusForUser(
+                    currentUser.id,
+                    clickedRole.name,
+                    !clickedRole.isActive
+                );
+                setErrorMessage('');
+            } catch (e) {
+                setIsActive((previous) => !previous);
+                const error = e as HttpClientRequestFailedError<FusionApiHttpErrorResponse>;
+                if (
+                    error.statusCode === 400 ||
+                    error.statusCode === 404 ||
+                    error.statusCode === 403
+                ) {
+                    setErrorMessage(error.response.error.message);
+                } else {
+                    setErrorMessage('Something went wrong when toggling the role');
+                }
+            }
+        },
+        [role]
+    );
+
+    const errorClassNames = classNames(styles.small, { [styles.error]: errorMessage });
+
+    return (
+        <div key={role.name} className={styles.roleManagementContainer}>
+            <div className={styles.row}>
+                <div>{role.displayName}</div>
+                {role.onDemandSupport && showSwitch && (
+                    <div>
+                        <Switch
+                            active={isActive}
+                            disabled={!role.onDemandSupport}
+                            onChange={() => toggleRoleStatus(role)}
+                        />
+                    </div>
+                )}
+            </div>
+            <div className={styles.row}>
+                <div className={styles.small}>{role.name}</div>
+                {role.activeToUtc && showSwitch && !errorMessage && (
```

Comment:
this works, but hard to read/understand, pref assing to a variable like: `const showExpireDate = role.activeToUtc && showSwitch && !errorMessage;` or into a memo